app/models/decision.rb

```diff
@@ -32,9 +32,9 @@ def weight_split_fully_to_variant
   end
 
   def reassign_existing_assignments
-    transaction do
-      bulk_assignment.save!
-      BulkReassignment.create!(assignments: existing_assignments, bulk_assignment: bulk_assignment)
+    bulk_assignment.save!
+    existing_assignments.find_in_batches do |subset|
```

Comment:
This isn't to improve performance in web context - it'll possibly actually be slower in web, but the web thread will time out. This is actually to be run in a background context without a wrapping transaction, and then we'll get reasonable throughput and no blocking except for the batch currently under consideration for as long as that txn takes to commit, presumably well sub-second.